pkg/apis/batch/v1alpha1/job.go

```diff
@@ -107,6 +107,34 @@ type JobSpec struct {
 	// +kubebuilder:validation:Minimum=1
 	// +optional
 	MinSuccess *int32 `json:"minSuccess,omitempty" protobuf:"varint,12,opt,name=minSuccess"`
+
+	// NetworkTopologies defines the NetworkTopologies config, this field works in conjunction with network topology feature and hyperNode CRD.
+	// +optional
+	NetworkTopologies *NetworkTopologiesSpec `json:"networkTopologies,omitempty" protobuf:"bytes,13,opt,name=networkTopologies"`
+}
+
+type NetworkTopologyMode string
+
+const (
+	// HardNetworkTopologyMode represents a strict network topology constraint that jobs must adhere to.
+	HardNetworkTopologyMode NetworkTopologyMode = "hard"
+
+	// SoftNetworkTopologyMode represents a flexible network topology constraint that allows jobs
+	// to cross network boundaries under certain conditions.
+	SoftNetworkTopologyMode NetworkTopologyMode = "soft"
+)
+
+type NetworkTopologiesSpec struct {
+	// Mode specifies the mode of the network topology constrain.
+	// +kubebuilder:validation:Enum=hard;soft
```

Comment:
Maybe move `+kubebuilder:validation:Enum=hard;soft` to L115

Reply:
Good idea！